languages/go/exercises/concept/using-map/.meta/design.md

```diff
@@ -0,0 +1,32 @@
+# Design
+
+## Goal
+
+The goal is to introduce the student to maps.
+
+## Learning objectives
+
+- initializing maps, empty and with values.
+- nil map will panic
+- settings key/value pairs
+- reading a key, non-existent key returns zero value of value type
+- using ok to check if the key existed
+
+## Out of scope
+
+- iterate over map
+- `map` data race
+
+## Concepts
+
+- `maps`:
+
+## Prequisites
+
+- `numbers`
+- `conditionals`
+- `functions`
```

Comment:
The prerequisites are a bit old. We can remove `numbers` as all basic types are introduced in the first exercise `types`.

```suggestion
- `types`
- `conditionals`
- `functions`
```

Now the big question is what to do with the `range` loop. We cannot really introduce the range loop without slices/maps. And to have `slices` -> `range` -> `maps is a bit problematic as we might have the same notion in the `slices` exercise that we want to range over a slice there, too.

I'd say we need to remove the need of `range` in this exercise. Sadly that kind of destroys the nice story. 😢

Reply:
Ah yeah, my bad, wasn't really paying attention to the concept map. I'd say we close this PR, and maybe on weekend I'll work on a new PR for this topic. 😁

Reply:
I don't mind if it stays open... We'll get it right eventually...

Reply:
> And to have slices -> range -> maps is a bit problematic as we might have the same notion in the slices exercise that we want to range over a slice there, too.

I don't know Go, but couldn't you have slices and range in the same Concept exercise? For some Concepts it can make sense to include two of them in a single exercise.

Reply:
could be done although I think there is enough to talk about each of them to have a separate exercise. The problem though is that range loops only make sense with slices and maps already introduced. But they are used here already.